fedmsg/meta/__init__.py

```diff
@@ -135,19 +137,23 @@ def msg2processor(msg, **config):
         return processors[-1]  # DefaultProcessor
 
 
-def legacy_condition(cls):
+def graceful(cls):
+    """ A decorator to protect against message structure changes.
+
+    Many of our processors expect messages to be in a certain format.  If the
+    format changes, they may start to fail and raise exceptions.  This decorator
+    is in place to catch and log those exceptions and to gracefully return
+    default values.
+    """
     def _wrapper(f):
-        def __wrapper(msg, legacy=False, **config):
+        @functools.wraps(f)
+        def __wrapper(msg, **config):
             try:
                 return f(msg, **config)
-            except KeyError:
-                if legacy:
-                    return cls()
-                else:
-                    raise
+            except Exception:
```

Comment:
I'd prefer to not handle such a broad exception class here. ``KeyError`` seems appropriate.

If, for example, anyone wants to make an exception class for meta-related problems, this will need to change and people will have started relying (unknowingly) on this. It also looks like this needs test coverage.